cmd/up.go

```diff
@@ -42,8 +52,197 @@ func newUpCmd() *cobra.Command {
+			return err
+		case expectNop && changes != nil && changes.HasChanges():
+			return errors.New("error: no changes were expected but changes occurred")
+		default:
+			return nil
+		}
+	}
+
+	// up implementation used when the source of the Pulumi program is a URL.
+	upURL := func(url string, opts backend.UpdateOptions) error {
+		if !workspace.IsTemplateURL(url) {
+			return errors.Errorf("%s is not a valid URL", url)
+		}
+
+		// Retrieve the template repo.
+		repo, err := workspace.RetrieveTemplates(url, false)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			contract.IgnoreError(repo.Delete())
+		}()
+
+		// List the templates from the repo.
+		templates, err := repo.Templates()
+		if err != nil {
+			return err
+		}
+
+		// Make sure only a single template is found.
+		// Alternatively, we could consider prompting to choose one instead of failing.
+		if len(templates) != 1 {
+			return errors.Errorf("more than one application found at %s", url)
+		}
+		template := templates[0]
+
+		// Create temp directory for the "virtual workspace".
+		temp, err := ioutil.TempDir("", "pulumi-up-")
+		if err != nil {
+			return err
+		}
+		defer func() {
+			contract.IgnoreError(os.RemoveAll(temp))
+		}()
+
+		// Cleanup the project name/description if needed.
+		projectName := template.ProjectName
+		if projectName == "${PROJECT}" {
+			projectName = workspace.ValueOrSanitizedDefaultProjectName(projectName, template.Name)
+		}
+		projectDescription := template.ProjectDescription
+		if projectDescription == "${DESCRIPTION}" {
+			projectDescription = ""
+		}
+
+		// Copy the template files from the repo to the temporary "virtual workspace" directory.
+		if err = template.CopyTemplateFiles(temp, true, projectName, projectDescription); err != nil {
+			return err
+		}
+
+		// Change the working directory to the "virtual workspace" directory.
+		if err = os.Chdir(temp); err != nil {
+			return errors.Wrap(err, "changing the working directory")
+		}
+
+		// Get a stack, but don't set it to current.
+		s, err := requireStack(stack, true, opts.Display, false /*setCurrent*/)
+		if err != nil {
+			return err
+		}
+
+		// Get the existing config. stackConfig will be nil if there wasn't a previous deployment.
+		stackConfig, err := backend.GetLatestConfiguration(commandContext(), s)
+		if err != nil && err != backend.ErrNoPreviousDeployment {
+			return err
+		}
+
+		// Get the existing snapshot.
+		snap, err := s.Snapshot(commandContext())
+		if err != nil {
+			return err
+		}
+
+		// Handle config.
+		// If this is an initial preconfigured empty stack (i.e. configured in the Pulumi Console),
+		// use its config without prompting.
+		// Otherwise, use the values specified on the command line and prompt for new values.
+		// If the stack already existed and had previous config, those values will be used as the defaults.
+		var c config.Map
+		templateURL := getPreconfiguredEmptyStackTemplateURL(stackConfig, snap)
+		if templateURL != "" {
```

Comment:
There's a bug here I need to fix.

The scenario: you click a "Deploy with Pulumi" button on a README.md, which would take you to a landing page in the Pulumi Console that lets you create a new stack and fill out the config on that page. That page would be responsible for creating the stack, populating it with an initial empty deployment (which just contains the config that was filled out on the page) and then provides the user with the `pulumi up <url> -s <stack>` command to copy/paste to run. When that command runs, we need the CLI to see that the stack already exists, has an empty deployment that contains config that satisfies the template's config requirements, and contains a `pulumi:template` config value containing a matching URL. In which case we can just use that config, without prompting from the CLI.

I need to fix the code to check to make sure the URL specified on the command line matches the URL stored in the `pulumi:template` config value, and that the rest of the config from the stack satisfies the config requirements of the template. Right now it's just blindly accepting it.